src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -620,15 +620,25 @@ protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessa
                 return Task.FromCanceled<HttpResponseMessage>(cancellationToken);
             }
 
-            HttpMessageHandlerStage handler = _handler ?? SetupHandlerChain();
```

Comment:
Should the synchronous version also move this after the validation for consistency?

Reply:
Re-ordered the validation.